### PR TITLE
filter out service accounts from members

### PR DESCRIPTION
**Pull request by zreigz**
**What this PR does / why we need it**: filter out service accounts from members

```release-note
filter out service accounts from the member's list
```

### CODE REVIEW

**Comment by zreigz** [comment]
@p0lyn0mial PTAL

File: api/pkg/provider/kubernetes/member.go [hunk 1]
```diff
@@ -77,9 +79,22 @@ func (p *ProjectMemberProvider) List(userInfo *provider.UserInfo, project *kuber
...
+	}
+
 	projectMembers := []*kubermaticapiv1.UserProjectBinding{}
 	for _, member := range allMembers {
 		if member.Spec.ProjectID == project.Name {
+
+			// skip if user is service account type
+			if strings.HasPrefix(member.Spec.UserEmail, serviceAccount) {
+				userEmailParts := strings.Split(member.Spec.UserEmail, "@")
+				if _, err := masterImpersonatedClient.Users().Get(userEmailParts[0], metav1.GetOptions{}); err == nil {
```
Comment (p0lyn0mial) [inline api/pkg/provider/kubernetes/member.go]: why do we have send `GET` request ?

File: api/pkg/provider/kubernetes/member_test.go [hunk 2]
```diff
@@ -61,81 +62,65 @@ func TestListBinding(t *testing.T) {
...
-					ObjectMeta: metav1.ObjectMeta{
-						Name: "differentProjectBinding",
-					},
-					Spec: kubermaticv1.UserProjectBindingSpec{
-						ProjectID: createProject("abcd").Name,
-						UserEmail: "bob@acme.com",
-						Group:     fmt.Sprintf("owners-%s", createProject("abcd").Name),
-					},
-				},
+				createBinding("abcdBinding", "my-first-project-ID", "bob@acme.com", "owners"),
```
Comment (p0lyn0mial) [inline api/pkg/provider/kubernetes/member_test.go:73]: sweet !

File: api/pkg/provider/kubernetes/member_test.go [hunk 3]
```diff
@@ -61,81 +62,65 @@ func TestListBinding(t *testing.T) {
...
-					ObjectMeta: metav1.ObjectMeta{
-						Name: "cdBinding",
-					},
-					Spec: kubermaticv1.UserProjectBindingSpec{
-						ProjectID: createProject("1234").Name,
-						UserEmail: "bob@acme.com",
-						Group:     fmt.Sprintf("owners-%s", createProject("123").Name),
-					},
-				},
+				createBinding("abcdBinding", "my-first-project-ID", "bob@acme.com", "owners"),
```
Comment (p0lyn0mial) [inline api/pkg/provider/kubernetes/member_test.go:78]: thanks for refactoring this :)

File: api/pkg/provider/kubernetes/member.go (hunk 1 above)
Comment (zreigz) [reply api/pkg/provider/kubernetes/member.go]: The email address is not enough to be sure if it SA or regular user. `Get` checks if it's a user with an email like SA and it should be displayed or real SA and must be omitted.
The user can have an email like "serviceaccount-bob@loods.com" but it's not SA.

File: api/pkg/provider/kubernetes/member.go [hunk 4]
```diff
@@ -80,6 +86,14 @@ func (p *ProjectMemberProvider) List(userInfo *provider.UserInfo, project *kuber
 	projectMembers := []*kubermaticapiv1.UserProjectBinding{}
 	for _, member := range allMembers {
 		if member.Spec.ProjectID == project.Name {
+
+			// skip if user is service account type
+			if strings.HasPrefix(member.Spec.UserEmail, serviceAccount) {
+				userEmailParts := strings.Split(member.Spec.UserEmail, "@")
```
Comment (p0lyn0mial) [inline api/pkg/provider/kubernetes/member.go:94]: should we check the length of `userEmailParts` slice before processing ? The next line can blow if the slice is empty.

File: api/pkg/provider/kubernetes/member.go [hunk 5]
```diff
@@ -80,6 +86,14 @@ func (p *ProjectMemberProvider) List(userInfo *provider.UserInfo, project *kuber
 	projectMembers := []*kubermaticapiv1.UserProjectBinding{}
 	for _, member := range allMembers {
 		if member.Spec.ProjectID == project.Name {
+
+			// skip if user is service account type
```
Comment (p0lyn0mial) [inline api/pkg/provider/kubernetes/member.go]: please add a comment why we are doing this.

File: api/pkg/provider/kubernetes/member.go (hunk 4 above)
Comment (zreigz) [reply api/pkg/provider/kubernetes/member.go:94]: yes, I can add the validation but generally, it shouldn't happen.
The email is generated in the code for SA (should be always correct), for members is already validated:
https://github.com/kubermatic/kubermatic/blob/master/api/pkg/handler/v1/user/user.go#L276

File: api/pkg/provider/kubernetes/member.go (hunk 5 above)
Comment (zreigz) [reply api/pkg/provider/kubernetes/member.go]: done

**Comment by zreigz** [comment]
@p0lyn0mial PTAL

**Pull request merged**